### PR TITLE
Initiate data integration report template

### LINKED ISSUE #351: Initiate data integration html report
**Please provide some background on the proposed additions or changes.**

We want to add a html report template to the data integration module, similar to what we have in the clustering module.

**What are the changes that you are proposing?**

We should initiate the data integration html report template with the following:
- a table displaying the size of the dataset (number of cells in each library)
- UMAPs displaying the unintegrated and the integrated (for each of the provided methods) data

**Please describe the proposed solution.**

A `integration-report-template.Rmd` should be added to the `optional-integration-analysis` module to do the following:
- take the integrated SCE object as input
- generate a table with the number of cells within each library that has been integrated
- generate UMAPs with the unintegrated and integrated data

### PR DESCRIPTION
**Pull request by cbethell**
**Issue Addressed**
Closes #351

**What is the purpose of these changes?**

This PR initiates the data integration report template.

**What changes did you make?**

A `integration-report-template.Rmd` file was added to the `optional-integration-analysis` module. This template currently contains the following:
- a breakdown of number of batches and number of cells per batch
- UMAPs for each of the integration methods being compared (including unintegrated)

**Were there any other solutions that you tried?**

It looks like we may need to tweak the `perform-integration.R` script here as well, as I tried running the template using both `"fastMNN, harmony"` as the integration methods but only the harmony results were found in the integrated SCE object

**Any comments, concerns, or questions important for reviewers**
- Does this tackle #351 as you would expect?

**Checklist**

Place an `x` in all boxes that you have completed.

- [x] I have run the most recent version of the code
- [ ] If I am adding in reports or generating any plots, I have reviewed the necessary reports
- [ ] I have added all necessary documentation (if applicable)

### CODE REVIEW

**Comment by cbethell** [comment]
Okay I added the `perform_dim_reduction()` function as we discussed @allyhawkins and that fixed the missing dimension reduction results issue -- here is a zip of the rendered html as it currently is:
[integration-report-template.html.zip](https://github.com/AlexsLemonade/scpca-downstream-analyses/files/11900443/integration-report-template.html.zip)

File: optional-integration-analysis/integration-report-template.Rmd [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+cell_label_column <- params$batch_column
+batch_umaps <- integration_methods %>%
+  purrr::map(~ plot_integration_umap(sce = integrated_sce,
+                                     integration_method = .x,
+                                     cell_label_column = params$batch_column,
+                                     legend_title = "Batches",
+                                     seed = params$seed))
+batch_legend <- cowplot::get_legend(batch_umaps[[1]])
+  
+ggpubr::ggarrange(plotlist = batch_umaps,
```
[the remark this answers is not in the document]
Comment (allyhawkins) [reply optional-integration-analysis/integration-report-template.Rmd:115]: I think you are missing the `legend_labels` argument when you call the umap plotting function: 
https://github.com/AlexsLemonade/sc-data-integration/blob/30f200c12e7259590cc12d571837e632bc92a8aa/analysis_templates/01-single-group-integration-check-template.Rmd#L411C54-L411C71

**Review by allyhawkins — COMMENTED** [review]
Thanks for getting this started. I added a number of comments to help clean things up. The biggest one is to update what is currently `perform_dim_reduction` to do a little more work.

**Comment by cbethell** [comment]
Implemented those comments you left @allyhawkins, thank you! I also successfully tested the `gt` package installation in the report template as we discussed 👍

File: utils/integration-functions.R [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+plot_integration_umap <- function(sce,
+                                  integration_method,
+                                  cell_label_column,
+                                  include_legend_counts = TRUE,
+                                  legend_title = NULL, 
+                                  legend_labels = NULL,
+                                  plot_colors = NULL,
+                                  plot_title = NULL,
+                                  seed = NULL) {
+
```
Comment (allyhawkins) [inline utils/integration-functions.R:17]: Can we get some more documentation on these functions? Similar to the format we have used for other `utils`? https://github.com/AlexsLemonade/scpca-downstream-analyses/blob/cbcea03855d89a93deb7d14621fecf85a06e9250/utils/clustering-functions.R#L12-L21

**Review by allyhawkins — COMMENTED** [review]
This looks mostly good, just a few more lingering comments.

File: utils/integration-functions.R [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+plot_integration_umap <- function(sce,
+                                  integration_method,
+                                  cell_label_column,
+                                  legend_title = NULL, 
+                                  legend_labels = NULL,
+                                  plot_colors = NULL,
+                                  plot_title = NULL,
+                                  seed = NULL) {
+  
+  # Purpose: Generate UMAP plots using integration results from a SingleCellExperiment object
```
Comment (allyhawkins) [inline utils/integration-functions.R:17]:
```suggestion
  # Purpose: Generate UMAP plots for a merged or integrated SingleCellExperiment object containing multiple batches
```
Technically this is used for unintegrated datasets, but the key is that we are plotting the batch labels so there needs to be more than one batch present.

**Review by allyhawkins — APPROVED** [review]
Looks good just one minor comment!

**Pull request merged**